src/crimson/os/seastore/segment_manager/zbd.cc

```diff
@@ -200,6 +212,31 @@ static seastar::future<size_t> get_zone_capacity(
   );
 }
 
+static seastar::future<size_t> get_nr_cnv_zones(
+  seastar::file &device,
+  uint32_t nr_zones)
+{
+  return seastar::do_with(
+    ZoneReport(nr_zones),
+    [&](auto &zr) {
+	size_t cnv_zones = 0;
```

Comment:
cnv_zones cannot live across the continuations, move to where it is actually used.

Reply:
Capturing by reference won't extend its life, the variable on stack is only valid within the continuation.